RFP-01.md

```diff
@@ -0,0 +1,169 @@
+    /**
+     * View function for determining the backend URI for the sub-pool
+     * @return string URI
+     */
+    function backendUri(SubPool pool) external view returns (string memory);
+
+    // --- permissionless functions ---
+    /**
+     * Deploy a `SubPool` deterministically based on `msg.sender`.
+     * @notice The `msg.sender` SHOULD NOT be a contract.
+     * @param token The nominated token to use for collateral
+     * @param amt How much collateral to add to the sub-pool (must be minimally collateralised)
+     * @param cowAmt Amount of COW to add as collateral
+     * @param backendUri The URI to send the batches to the solver
+     * @dev Emits an event `SubPoolCreated` when creating a sub-pool
+     */
+    function create(IERC20 token, uint256 amt, uint256 cowAmt, string calldata backendUri) external;
+
+    // --- cow dao bonding pool authed functions ---
+
+    /**
+     * Set the delay for a sub-pool to exit
+     * @dev Emits an event when setting a delay
+     * @param delay The delay in seconds
+     */
+    function setExitDelay(uint256 delay) external auth;
+    /**
+     * Allow `token` to be used as collateral for the sub-pool.
+     * @dev Emits an event `AllowCollateral` when allowing/updating a token
+     */
+    function allowCollateral(IERC20 token, uint256 min) external auth;
+    /**
+     * Revoke `token` as a collateral for sub-pools
+     * @dev Emits an event `RevokeCollateral` when revoking a token
+     */
+    function revokeCollateral(IERC20 token) external auth;
+    /**
+     * Fine a sub-pool
+     * @param pool Which sub-pool to fine
+     * @param tokens The tokens to fine (sending to `msg.sender`)
+     * @param amts The amount of tokens to fine
+     * @dev Emits an event `Fine` when fining a sub-pool
+     */
+    function fine(SubPool pool, IERC20[] calldata tokens, uint256[] calldata amts) external auth;
+
+    // --- SubPool authed functions ---
+
+    /**
+     * Indicate a sub-pool's intent to quit the batch auction
+     * At this point the solver whose sub-pool it is is now removed from
+     * the competition.
+     * This then starts the `delay` timer until exit is allowed.
+     * @param pool The sub-pool to announce an exit for
+     * @dev Emits an event `AnnounceExit`
+     */
+    function announceDestroy(SubPool pool) external onlySubpoolAuth;
+    /**
+     * Withdraw any remaining collateral / cow from the sub-pool.
+     * Funds are sent to `msg.sender`.
+     * @dev Should emit an event `SubPoolDestroyed` on the factory
+     */
+    function finalizeExit(SubPool pool) external onlySubpoolAuth;
+    /**
+     * Set the backend URI for the sub-pool
+     * @param pool The sub-pool to set the URI for
+     * @param newUri The new URI to set
+     * @dev Emits an event `SetBackendUri` on the factory when setting a URI
+     */
+    function setBackendUri(SubPool pool, string calldata newUri) external onlySubpoolAuth;
```

Comment:
nit: maybe this doesn't need auth if it implies you can only update your own pool's uri anyway (could compute pool from message.sender)

Reply:
See previous comment on multiple owners